packages/react/src/ActionList/TrailingAction.tsx

```diff
@@ -0,0 +1,59 @@
+import React, {forwardRef} from 'react'
+import Box from '../Box'
+import {Button, IconButton} from '../Button'
+import type {ForwardRefComponent as PolymorphicForwardRefComponent} from '../utils/polymorphic'
+
+export type ActionListTrailingActionProps = {
+  as?: 'button' | 'a'
+  href?: string
```

Comment:
What would you think about defining types like this?

```tsx
{
  as?: 'button'
  href: never // href is invalid for button type
} | 
{
  as?: 'a'
  href: 'string' // href is required for a type
}
```

I have seen we did this before and I thought it is cool!

Reply:
OOH TIL 😁

Reply:
ok this didn't quite work for me 😅

Reply:
@broccolinisoup, I continued to poke around with the syntax. See https://github.com/primer/react/commit/194f847351b216644a52d757e3a98a8c49c02b5d. 
I thought I was close, but typescript gets mad at the `href = null` thats passed into line 13 because `button` is assumed to be the default. 

I looked at other examples in the code and did not see any examples that achieve what we want. 🤔 

I would be open any ideas you have to get the types to work, but I don't think this should block us from moving forward!